lib/licensed/sources/npm.rb

```diff
@@ -47,7 +47,9 @@ def packages
       # Recursively parse dependency JSON data.  Returns a hash mapping the
       # package name to it's metadata
       def recursive_dependencies(dependencies, result = {})
+        yarn_lock_present = File.exist?(@config.pwd.join("yarn.lock"))
```

Comment:
Could you move this to a separate method and cache the result so that the file isn't looked at for every call to `recursive_dependencies`? :bow: